src/network.c

```diff
@@ -851,3 +851,24 @@ int setup_SSL_connection(Connection *conn, SSL_CTX *ctx) {
   } while (ssl_ret != 1);
   return 0;
 }
+
+/**
+ * Sets 5 minute read and write timeouts on the passed-in socket.  Calls exit()
+ * if setsockopt fails for any reason.
+ * @param sockfd the socket to modify
+ */
+void set_socket_timeout_or_die(int sockfd) {
+  struct timeval timeout;
+  timeout.tv_sec = 300;
```

Comment:
Out of curiosity, why 5 minutes? That seems like a long time to me. Are there cases where a process should last much more than 30-60 seconds?